File: ios-app/NailsDashIOS/Sources/Core/Auth/AuthModels.swift
import Foundation

enum VerificationPurpose: String, Encodable {
    case register
    case login
    case resetPassword = "reset_password"
}

struct LoginRequest: Encodable {
    let phone: String
    let password: String
    let login_portal: String
}

struct RegisterRequest: Encodable {
    let phone: String
    let verification_code: String
    let username: String
    let password: String
    let full_name: String?
    let referral_code: String?
}

struct SendVerificationCodeRequest: Encodable {
    let phone: String
    let purpose: VerificationPurpose
}

struct SendVerificationCodeResponse: Decodable {
    let message: String
    let expires_in: Int
}

struct VerifyCodeRequest: Encodable {
    let phone: String
    let code: String
    let purpose: VerificationPurpose
}

struct VerifyCodeResponse: Decodable {
    let valid: Bool
    let message: String
}

struct TokenResponse: Decodable {
    let access_token: String
    let refresh_token: String
    let token_type: String
}

struct AuthUser: Decodable {
    let id: Int
    let phone: String
    let username: String
    let full_name: String?
    let email: String?
    let avatar_url: String?
    let gender: String?
    let date_of_birth: String?
    let phone_verified: Bool?
    let is_admin: Bool
    let store_id: Int?
}

File: android-app/app/src/main/java/com/nailsdash/android/data/model/AuthModels.kt
package com.nailsdash.android.data.model

enum class VerificationPurpose {
    register,
    login,
    reset_password,
}

data class LoginRequest(
    val phone: String,
    val password: String,
    val login_portal: String = "frontend",
)

data class RegisterRequest(
    val phone: String,
    val verification_code: String,
    val username: String,
    val password: String,
    val full_name: String? = null,
    val referral_code: String? = null,
)

data class SendVerificationCodeRequest(
    val phone: String,
    val purpose: VerificationPurpose,
)

data class SendVerificationCodeResponse(
    val message: String,
    val expires_in: Int,
)

data class VerifyCodeRequest(
    val phone: String,
    val code: String,
    val purpose: VerificationPurpose,
)

data class VerifyCodeResponse(
    val valid: Boolean,
    val message: String,
)

data class TokenResponse(
    val access_token: String,
    val refresh_token: String,
    val token_type: String,
)

data class RefreshTokenRequest(
    val refresh_token: String,
)

data class AuthUser(
    val id: Int,
    val phone: String,
    val username: String,
    val full_name: String? = null,
    val email: String? = null,
    val avatar_url: String? = null,
    val gender: String? = null,
    val date_of_birth: String? = null,
    val phone_verified: Boolean? = null,
    val is_admin: Boolean,
    val store_id: Int? = null,
)
